homeassistant/components/notify/clicksend.py

```diff
@@ -0,0 +1,71 @@
+"""
+Clicksend platform for notify component.
+For more details about this platform, please refer to the documentation at
+https://clicksend.com/help
+"""
+
+# Import dependencies.
+import logging
+import requests
+import json
+import base64
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.notify import (
+    PLATFORM_SCHEMA, BaseNotificationService)
+
+# Get logger instance.
+_LOGGER = logging.getLogger(__name__)
+
+# Set platform  parameters.
+CONF_API_URL = 'https://rest.clicksend.com/v3/sms/send'
+CONF_USERNAME = 'username'
+CONF_API_KEY = 'api_key'
+CONF_TO_NO = 'to_no'
+
+# Validate parameter schema.
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_TO_NO): cv.string,
+})
+
+# Define service instance.
+def get_service(hass, config, discovery_info=None):
+
+    # Set notification service instance.
+    return ClicksendNotificationService(config[CONF_USERNAME], config[CONF_API_KEY], config[CONF_TO_NO])
```

Comment:
Would be nice if you could check if the credentials are valid and make setup fails if not. Otherwise the users end-up with a non-working platform in their setup.